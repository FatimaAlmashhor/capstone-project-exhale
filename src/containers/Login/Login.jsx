import React, { useState, useEffect } from 'react';
import firebase from 'firebase';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';
import LoginForm from '../../components/LoginForm';
import { SIGNUP_ROUTE } from '../../routes';
import ResetPassword from '../../components/resetPassword/ResetPassword';
import Modal, { Header, Body } from '../../components/common/Modal/Modal';

const Login = () => {
  const [isSignedIn, setisSignedIn] = useState(false);
  const [show, setShow] = useState(false);

  const { t } = useTranslation();
  const location = useLocation();

  useEffect(() => {
    firebase.auth().onAuthStateChanged((user) => {
      setisSignedIn(!!user);
    });
  });

  const handleClose = () => setShow(false);

  return (
    <div>
      {isSignedIn ? null : (
        <>
          <div className="my-2 text-center border-b">
            <div className="inline-block px-2 text-sm font-medium leading-none tracking-wide text-gray-600 transform translate-y-1/2 bg-white">
              {t('with email')}
            </div>
          </div>
          <LoginForm />
          <button
            type="button"
            className="inline-block w-full text-sm font-normal text-center text-blue-500 align-baseline focus:outline-none hover:text-blue-700"
            onClick={() => {
              setShow(true);
            }}
          >
            {t('forgetpassword')}
          </button>
          {show ? (
            <div>
              <Modal show={show} onClick={handleClose}>
                <div>
                  <Header title={t('Reset')} onClick={handleClose} />
                  <Body>
                    <ResetPassword />
                  </Body>
                </div>
              </Modal>
            </div>
          ) : null}
          <p className="text-center text-gray-600 text-md">
            {t('haveAccount')}
            <Link
              className="px-2 text-blue-500 text-md focus:outline-none"
              to={{
                pathname: SIGNUP_ROUTE,
                state: { modal: location.state.modal },
              }}
            >
              {t('Create')}
            </Link>
          </p>
        </>
      )}
    </div>
  );
};

export default Login;
