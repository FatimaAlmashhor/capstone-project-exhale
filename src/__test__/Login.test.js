import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../containers/Login';

it('renders correctly', () => {
    const fun = function (params) {
        // 
    }
    const foo= true;
  const tree = renderer.create(<Login show={foo} onClick={fun}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
