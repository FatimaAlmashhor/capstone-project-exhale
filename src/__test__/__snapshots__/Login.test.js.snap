// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
    onClick={[Function]}
  >
    <div
      className="relative w-auto my-6 mx-auto max-w-5xl"
      onClick={[Function]}
    >
      <div
        className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none"
      >
        <div
          className="flex items-start justify-between p-5 border-b border-solid border-gray-300 rounded-t"
        >
          <h3
            className="text-3xl font-semibold"
          >
            SIGN IN
          </h3>
          <button
            className="p-1 ml-auto bg-transparent border-0 text-black opacity-5 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <span
              className="bg-transparent text-black opacity-5 h-6 w-6 text-2xl block outline-none focus:outline-none"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="relative p-6 flex-auto"
        >
          <div
            id="firebaseui_container"
          />
          <div
            className="my-2 border-b text-center"
          >
            <div
              className="leading-none px-2 inline-block text-sm text-gray-600 tracking-wide font-medium bg-white transform translate-y-1/2"
            >
              with email
            </div>
          </div>
          <div
            className="w-full max-w-xs mx-auto"
          >
            <form
              className="bg-white  rounded  pt-6 pb-3 mb-4"
              onSubmit={[Function]}
            >
              <input
                className=" appearance-none border rounded w-full p-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="yourEmail"
                type="text"
              />
              <input
                className=" mt-5 appearance-none border rounded w-full p-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="yourPassword"
                type="password"
              />
              <div
                className="flex items-center justify-between mb-6"
              >
                <button
                  className="mt-5 px-4 py-3 w-full rounded text-white inline-block shadow-lg bg-blue-800 hover:bg-blue-600 focus:bg-blue-700"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </form>
          </div>
          <button
            className="inline-block text-center w-fullfocus:outline-none  align-baseline font-normal text-sm text-blue-600 hover:text-blue-800"
            type="button"
          >
            forgetpassword
          </button>
          <p
            className=" text-center text-md  text-gray-600 "
          >
            haveAccount
            <button
              className="px-2 text-md text-blue-600 focus:outline-none"
              onClick={[Function]}
              type="button"
            >
              Create
            </button>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="opacity-25 fixed inset-0 z-40 bg-black"
  />
  ;
</div>
`;
